### PR TITLE
feat: Generate CRD docs

### PR DESCRIPTION
**Pull request by sergeyshevch**
Screenshot 2024-04-24 at 20 50 09

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -274,6 +280,10 @@ controller-gen: $(LOCALBIN)
 envtest: $(LOCALBIN)
 	@test -x $(ENVTEST) || GOBIN=$(LOCALBIN) go install sigs.k8s.io/controller-runtime/tools/setup-envtest@$(ENVTEST_VERSION)
 
+.PHONY: crd-ref-docs
+crd-ref-docs: $(LOCALBIN)
+	@test -x $(CRD_REF_DOCS) || GOBIN=$(LOCALBIN) go install github.com/elastic/crd-ref-docs@latest
```
Comment (hiddenmarten) [inline Makefile:285]: I think it's a good pattern to define a particular version where it's possible.
For inspiration take a look here:

https://github.com/aenix-io/etcd-operator/blob/cc22fbd58659b9cb6a225e825c4eb6c30e2247ea/Makefile#L316C1-L319C95

**Review by hiddenmarten — APPROVED** [review]
LGTM with minor comments

**Review by kvaps — APPROVED** [review]

**Review by sircthulhu — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (sergeyshevch) [reply Makefile:287]: Sadly it doesn't provide any way to fetch version from binary

**Review by kvaps — APPROVED** [review]
LGTM

**Review by Kirill-Garbar — APPROVED** [review]

**Pull request merged**